### DISCUSSION TITLE
need help in configuring

### DISCUSSION THREAD

**Discussion by deepakchandarr** [stuttter/ludicrousdb#161] [Help] [answered]
I want to split the database of multisite  into 100 dbs, db00, db01, so on upto db99 .and when a new blog gets added to the multisite, the last two digits of the blog id go to a specified database,
for example,
We can predict what database a blog is in by looking

at the last two digits of the id. Examples:
blog_id: 9, in db09
blog_id: 74, in db74
blog_id: 999989, in db89
blog_id: 9200, in db00
I almost set up everything as in the instruction but don't know how to configure this algorithm
when i try to do , i keep getting critical error

**Comment by JJJ** [comment, marked as answer]
The last-two-digits scheme can be expressed directly with modulo. Register datasets named `db00` through `db99`, then use a callback like this:

```php
$wpdb->add_callback(
	function ( $query, $wpdb ) {
		$pattern = '/^' . preg_quote( $wpdb->base_prefix, '/' ) . '(\d+)_/';

		if ( ! preg_match( $pattern, $wpdb->table, $matches ) ) {
			return null;
		}

		$blog_id = (int) $matches[1];

		return sprintf( 'db%02d', $blog_id % 100 );
	}
);
```

That maps blog 9 to `db09`, 74 to `db74`, 999989 to `db89`, and 9200 to `db00`. Returning `null` leaves network-global tables on the default dataset.

Every dataset the callback can return must already be added with `$wpdb->add_database()`, and every site's complete table set must already exist in the corresponding database. LudicrousDB only routes queries; it does not create or move the tables. Also keep the divisor fixed after data is placed, because changing it remaps existing sites.

I have added this pattern, along with the migration cautions, to the wiki's Partitioning Recipes page.
